src/test/java/pl/piekoszek/gorskimatches/controllers/EquationControllerTest.java

```diff
@@ -0,0 +1,42 @@
+package pl.piekoszek.gorskimatches.controllers;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class EquationControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    public void shouldReturnCorrectFormat() throws Exception {
+        var equation = mockMvc.perform(get("/api/equation/random")).andExpect(status().isOk()).andReturn().getResponse().getContentAsString();
+        assertThat(equation.matches("\\d[-+]\\d=\\d")).isTrue();
+    }
+
+    @Test
+    public void shouldReturnTrue() throws Exception {
+        var eq = mockMvc.perform(post("/api/equation/solution").content("""
```

Comment:
eq would be good for nice language like Go, but we're learning Java, so use `equation` ;)